Tests/MailgunTests/MailgunClientTests.swift

```diff
@@ -0,0 +1,49 @@
+import XCTest
+import Mail
+import SMTP
+import Mailgun
+@testable import Vapor
+
+
+class MailgunClientTests: XCTestCase {
+    static let allTests = [
+        ("testProvider", testProvider),
+    ]
+
+    let apiKey = "MG.YOUR_KE" // Set here, but don't commit to git!
+    let domain = "MG.YOUR_DOMAIN" // Set here, but don't commit to git!
+
+    func testProvider() throws {
+        if apiKey == "MG.YOUR_KEY" {
+            print("Not testing Mailgun as no API Key is set")
+            return
+        }
+        if apiKey == "MG.YOUR_DOMAIN" {
+            print("Not testing Mailgun as no domain is set")
+            return
+        }
+        let config = Config([
+            "mailgun": [
+                "apiKey": Node(apiKey),
+                "domain": Node(domain)
+            ],
+        ])
+        let drop = try makeDroplet(config: config)
+        try drop.addProvider(Mail.Provider<MailgunClient>.self)
+
+        let email = SMTP.Email(from: "anth-mail-from@mailinator.com",
```

Comment:
I have just noticed that I used `bygri-mail` in Sendgrid which is not ideal. Let's change this to `vapor-mailgun-from` and `vapor-mailgun`, and I'll change Sendgrid to `vapor-sendgrid-from` and `vapor-sendgrid` instead.